### PR TITLE
[HLSL] Fix resource kind for RasterizerOrderedStructuredBuffer

### PR DESCRIPTION
**Pull request by bogner**
This is a kind of StructuredBuffer, so it should be "Raw" and not "Typed".

### CODE REVIEW

**Review by llvm-beanz — APPROVED** [review]

File: clang/lib/Sema/HLSLExternalSemaSource.cpp [hunk 1]
```diff
@@ -480,8 +480,8 @@ void HLSLExternalSemaSource::defineHLSLTypesWithForwardDeclarations() {
 
   onCompletion(Decl, [this](CXXRecordDecl *Decl) {
     setupBufferType(Decl, *SemaPtr, ResourceClass::UAV,
-                    ResourceKind::TypedBuffer,
-                    /*IsROV=*/false, /*RawBuffer=*/false)
+                    ResourceKind::TypedBuffer, /*IsROV=*/false,
+                    /*RawBuffer=*/false)
```
Comment (damyanp) [inline clang/lib/Sema/HLSLExternalSemaSource.cpp:484]: Why was this reformatted? I think there's no change here other than formatting - I thought we had something that enforced clang-format conformance?

File: clang/lib/Sema/HLSLExternalSemaSource.cpp [hunk 2]
```diff
@@ -547,9 +543,8 @@ void HLSLExternalSemaSource::defineHLSLTypesWithForwardDeclarations() {
              .addSimpleTemplateParams(*SemaPtr, {"element_type"})
              .Record;
   onCompletion(Decl, [this](CXXRecordDecl *Decl) {
-    setupBufferType(Decl, *SemaPtr, ResourceClass::UAV,
-                    ResourceKind::TypedBuffer, /*IsROV=*/true,
-                    /*RawBuffer=*/true)
+    setupBufferType(Decl, *SemaPtr, ResourceClass::UAV, ResourceKind::RawBuffer,
+                    /*IsROV=*/true, /*RawBuffer=*/true)
```
Comment (damyanp) [inline clang/lib/Sema/HLSLExternalSemaSource.cpp:547]: When would we ever want RawBuffer to be false and ResourceKind to be something other than RawBuffer? Or are we going to be removing one of these in the future?

**Review by damyanp — APPROVED** [review]

File: clang/lib/Sema/HLSLExternalSemaSource.cpp (hunk 1 above)
Comment (bogner) [reply clang/lib/Sema/HLSLExternalSemaSource.cpp:484]: `clang-format` was complaining about the line I actually changed and I noticed that all of these looked funny. Seems like there's a "comment at the beginning of a line" heuristic that stops clang-format from messing with these, but if you collapse them to a single line and then run clang-format it does something pretty reasonable.

File: clang/lib/Sema/HLSLExternalSemaSource.cpp (hunk 2 above)
Comment (bogner) [reply clang/lib/Sema/HLSLExternalSemaSource.cpp:547]: `ResourceKind` won't be specified here in the future (See #104862), which will remove the redundancy. This change doesn't actually do much as is, but the inconsistency was confusing.

**Pull request merged**